tutorials/simulation_exporting.ipynb

```diff
@@ -0,0 +1,326 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #15.            matrix = [self._nd_subdomain()]
</pre><p>I see the benefits of using mixin-concepts, but I really don't like this referal to a non-existent attribute. While there should be little room for confusion in this instance, thus it may be acceptable, it goes against all principles of object oriented design. The 'correct' solution is to move the methods to a superclass, or to restructure the code in a different way. Also, we should be cautious to put such code in a tutorial, since they are naturally read as establishing good practice. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/708/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Reply:
Hmm. I see what you mean. But according to this [post](https://www.pythontutorial.net/python-oop/python-mixin/), a mixin may call non-implemented methods.

Reply:
I see the point - let's go with this.